rosidl_generator_cpp/resource/msg__struct.hpp.template

```diff
@@ -144,6 +152,14 @@ for field in spec.fields:
   typedef std::weak_ptr<@(cpp_full_name)<ContainerAllocator> const>
     ConstWeakPtr;
 
+  // pointer types similar to ROS 1, use SharedPtr / ConstSharedPtr instead
+  typedef @(deprecated_macro_name)
+    std::shared_ptr<@(cpp_full_name)<ContainerAllocator>>
+    Ptr;
+  typedef @(deprecated_macro_name)
+    std::shared_ptr<@(cpp_full_name)<ContainerAllocator> const>
+    ConstPtr;
```

Comment:
Looking at this:

https://gcc.gnu.org/onlinedocs/gcc-3.3/gcc/Type-Attributes.html

It looks like the deprecation attribute should go at the end of the typedef. Maybe this doesn't matter and has to be this way on Windows? Have you tried it?